browser/resources/settings/brave_leo_assistant_page/brave_leo_assistant_page.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {PolymerElement} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js';
+import {WebUiListenerMixin} from 'chrome://resources/cr_elements/web_ui_listener_mixin.js';
+import {PrefsMixin} from 'chrome://resources/cr_components/settings_prefs/prefs_mixin.js';
+import {I18nMixin} from 'chrome://resources/cr_elements/i18n_mixin.js';
+import {getTemplate} from './brave_leo_assistant_page.html.js'
+import {BraveLeoAssistantBrowserProxy, BraveLeoAssistantBrowserProxyImpl}
+  from './brave_leo_assistant_browser_proxy.js'
+
+
+const BraveLeoAssistantPageBase =
+  WebUiListenerMixin(I18nMixin(PrefsMixin(PolymerElement)))
+
+/**
+ * 'settings-brave-leo-assistant-page' is the settings page containing
+ * brave's Leo Assistant features.
+ */
+class BraveLeoAssistantPageElement extends BraveLeoAssistantPageBase {
+    static get is() {
+        return 'settings-brave-leo-assistant-page'
+    }
+
+    static get template() {
+        return getTemplate()
+    }
+
+    static get properties() {
+      return {
+        leoAssistantShowOnToolbarPref_: {
+          type: Boolean,
+          value: false,
+          notify: true,
+        },
+      }
+    }
+
+    leoAssistantShowOnToolbarPref_: boolean
+    leoAssistantShowPromptsPref_: boolean
+    browserProxy_: BraveLeoAssistantBrowserProxy =
+      BraveLeoAssistantBrowserProxyImpl.getInstance()
+
+    onResetAssistanceData_() {
+      this.browserProxy_.reset().then((result) => {
+        if(result) {
+          window.alert(this.i18n('braveLeoAssistantResetConfirmed'))
+          this.getShowLeoAssistantIcon_()
+        }
+      })
+    }
+
+    override ready () {
+      super.ready()
+
+      this.browserProxy_.initLeoAssistant()
+
+      this.browserProxy_.getShowLeoAssistantIcon().then((result) => {
+        this.onShowLeoAssistantIcon_(result)
+      })
+
+      this.addWebUiListener('settings-brave-leo-assistant-changed',
+      () => { this.getShowLeoAssistantIcon_() })
+    }
+
+    itemPref_(enabled: boolean) {
+      return {
+        key: '',
+        type: chrome.settingsPrivate.PrefType.BOOLEAN,
+        value: enabled,
+      }
+    }
+
+    private getShowLeoAssistantIcon_() {
+      this.browserProxy_.getShowLeoAssistantIcon().then((result) => {
+        this.onShowLeoAssistantIcon_(result)
+      })
+    }
+
+    private onShowLeoAssistantIcon_(isLeoIconVisible: boolean) {
+      this.leoAssistantShowOnToolbarPref_ = isLeoIconVisible
+    }
+
+    onLeoAssistantShowOnToolbarChange_(e: any) {
+      e.stopPropagation()
+      this.browserProxy_
+      .setShowLeoAssistantIcon(!this.leoAssistantShowOnToolbarPref_)
+      .then((result) => {
+        if(result) {
```

Comment:
I think result indicating success/failure is a bit misleading here - if I was using the API, I'd assume it was the new value of whether the icon is being shown. Maybe if the set fails it should return the unchanged value?

Reply:
Changed it to not return any value, because we  have event listener `settings-brave-leo-assistant-changed` which is called each time the Leo Icon on the sidebar is added or removed, and new visibility value passed as parameter to the event handler.

https://github.com/brave/brave-core/pull/19212/commits/414fb6cd24fb8e0b6dfbf0eefabbcb9392065272#diff-525de7c635b1f78970d667a5985b7d99d69f083b25c5329312ad88091ec21704R81